README.md

```diff
@@ -1,7 +1,123 @@
+
+## Training ResNet50 models
+First of all change the directory to `train_and_eval/train`. While training, we dump  model ckpts, training logs and run command inside a folder. 
+
+### Training Fixed Feature Baseline
+
+```bash 
+export CUDA_VISIBLE_DEVICES=0,1
+
+python train_imagenet.py --config-file rn50_configs/rn50_40_epochs.yaml --model.fixed_feature=2048 --data.train_dataset=$WRITE_DIR/train_500_0.50_90.ffcv --data.val_dataset=$WRITE_DIR/val_500_uncompressed.ffcv --data.num_workers=12 --data.in_memory=1 --logging.folder=trainlogs --logging.log_level=1 --dist.world_size=2 --training.distributed=1 --lr.lr=0.425
+```
+
+### Training MRL model
+
+```bash 
+export CUDA_VISIBLE_DEVICES=0,1
+
+python train_imagenet.py --config-file rn50_configs/rn50_40_epochs.yaml --model.mrl=1 --data.train_dataset=$WRITE_DIR/train_500_0.50_90.ffcv --data.val_dataset=$WRITE_DIR/val_500_uncompressed.ffcv --data.num_workers=12 --data.in_memory=1 --logging.folder=trainlogs --logging.log_level=1 --dist.world_size=2 --training.distributed=1 --lr.lr=0.425
+```
+
+By default, we start nesting from 8 dimensions (That is 2^3). In case we want to nest representations from 16 dimensions, add an additional flag of 
+```
+--model.nesting_start=4
+```
+
+### Training MRL-E model
+
+```bash 
+export CUDA_VISIBLE_DEVICES=0,1
+
+python train_imagenet.py --config-file rn50_configs/rn50_40_epochs.yaml --model.efficient=1 --data.train_dataset=$WRITE_DIR/train_500_0.50_90.ffcv --data.val_dataset=$WRITE_DIR/val_500_uncompressed.ffcv --data.num_workers=12 --data.in_memory=1 --logging.folder=trainlogs --logging.log_level=1 --dist.world_size=2 --training.distributed=1 --lr.lr=0.425
+```
+
+By default, we start nesting from 8 dimensions (That is 2^3). In case we want to nest representations from 16 dimensions, add an additional flag of 
+```
+--model.nesting_start=4
+```
+## Evaluating Trained Models
+First of all change the directory to `train_and_eval/eval`. We have two different scripts to evaluate the Fixed Feature (FF) baseline and MRL models, respectively.
+
+### Evaluating FF baseline
+In paper we only consider fixed `nesting_list=[8, 16, 32, 64, 128, 256, 512, 1024, 2048]`, therefore we hardcode evaluating on those FF mdoels such that the name for each folder with ckpt is `sh=0_mh=0_nesting_start=3_fixed_feature={dim}/` Please change the name accordingly in while evaluating your models. Furthermore, make sure that launch directory has validation sets corresponding to each dataset (V1/V2/A/R/Sketch). To evaluate, run the following command. 
+
+```
+python eval_ff.py --path [path to ckpt folders] --dataset [V2/A/Sketch/R/V1]
+```
+If `dataset` flag is not passed, it will evaluate on standard ImageNet validation set. 
+
+### Evaluating MRL models 
+```
+python eval_MRL.py --path [path to weight checkpoint, need to be .pt file] --dataset [V2/A/Sketch/R/V1] (--efficient)
+```
+
+## [Evaluating pre-trained checkpoints](eval_ckpts/)
```

Comment:
pre-trained MRL checkpoints?

Reply:
Okay I think this is unintentionally pointing to w/o any training model. I meant already trained by us